java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1011,4 +1012,39 @@ public void zcard(BaseClient client) {
                 assertThrows(ExecutionException.class, () -> client.zcard("foo").get());
         assertTrue(executionException.getCause() instanceof RequestException);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void type(BaseClient client) {
+        String nonExistingKey = UUID.randomUUID().toString();
+        String stringKey = UUID.randomUUID().toString();
+        String listKey = UUID.randomUUID().toString();
+        String hashKey = UUID.randomUUID().toString();
+        String setKey = UUID.randomUUID().toString();
+        String zsetKey = UUID.randomUUID().toString();
+        String streamKey = UUID.randomUUID().toString();
+
+        assertEquals(OK, client.set(stringKey, "value").get());
+        assertEquals(1, client.lpush(listKey, new String[] {"value"}).get());
+        assertEquals(1, client.hset(hashKey, Map.of("1", "2")).get());
+        assertEquals(1, client.sadd(setKey, new String[] {"value"}).get());
+        assertEquals(1, client.zadd(zsetKey, Map.of("1", 2.)).get());
+
+        // use custom command until XADD is implemented
+        String[] args = new String[] {"XADD", streamKey, "*", "field", "value"};
+        if (client instanceof RedisClient) {
+            assertNotNull(((RedisClient) client).customCommand(args).get());
+        } else if (client instanceof RedisClusterClient) {
+            assertNotNull(((RedisClusterClient) client).customCommand(args).get());
```

Comment:
This assert never fails - you don't validate the content of `ClusterValue` here.

Reply:
Ooh.  Good point.